### PR TITLE
Sample console application

### PR DESCRIPTION
**Pull request by IevhenIkonnykov**
- initial commit with the first sample console application

### CODE REVIEW

File: CDP4-SDK-Samples/SampleApp.cs [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+    using CDP4Common.Types;
+    using CDP4Dal;
+    using CDP4Dal.DAL;
+    using CDP4Dal.Operations;
+    using CDP4ServicesDal;
+    using NLog;
+    using NLog.Config;
+    using NLog.Targets;
+
+    public class SampleApp
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SampleApp.cs]: Should be `Program`

File: CDP4-SDK-Samples/SampleApp.cs [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+    {
+        private static ISession _session;
+        private static Credentials _credentials;
+        private static bool _isRunning = true;
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SampleApp.cs]: for `private` fields we use the camelcase convention: `isRunning`

File: CDP4-SDK-Samples/SampleApp.cs [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+        private static Uri _uri;
+
+        // Commands
+        private const string Open = "open";
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SampleApp.cs]: for `private` fields we use camelCase even for `const`.

File: CDP4-SDK-Samples/SampleApp.cs [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+        private const string Close = "close";
+        private const string Restore = "restore";
+        private const string GetIteration = "get_iteration";
+        private const string PostPerson = "post_person";
+        private const string PostParameter = "post_parameter";
+        private const string PostPfsl = "post_pfsl";
+        private const string PostPfslReorder = "post_pfsl_reorder";
+        private const string RemoveParameter = "remove_parameter";
+
+        static void Main(string[] args)
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SampleApp.cs]: Move all business logic to another class to have all these fields and methods static. And just instantiate that new class from here.

**Review by alexatstariongroup — APPROVED** [review]
In principle all fine, but I would like you to refactor a bit all the logic to keep it out of `Main()` so that it is not all static.

**Review by alexatstariongroup — CHANGES_REQUESTED** [review]
Sorry wrong status.

**Review by lxatstariongroup — CHANGES_REQUESTED** [review]
In general, the Code style and Naming styles have some issues.

I don't think you're using ReSharper so here is a screenshot from my version about naming style:
image

Also, we prefix the use of every method/property/field with "this."

File: CDP4-SDK-Samples/SampleApp.cs (hunk 3 above)
Comment (lxatstariongroup) [reply CDP4-SDK-Samples/SampleApp.cs]: Not for private const !
image

File: CDP4-SDK-Samples/SimpleConsoleApp.cs [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+                {
+                    printSeparator();
+                    Console.WriteLine("Something went wrong. Sorry about that.");
+                    Console.WriteLine(ex.Message);
+                    printSeparator();
+                }
+            }
+        }
+
+        private void executeCommand(string command)
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SimpleConsoleApp.cs]: Yeah so adding to what @lxatrhea already said on naming conventions. Methods should have PascalCase -> `ExecuteCommand`.

File: CDP4-SDK-Samples/SimpleConsoleApp.cs [hunk 6]
```diff
@@ -0,0 +1,507 @@
...
+        {
+            var args = command.Trim().Split(" ");
+            switch (args[0])
+            {
+                case openConnection:
+                {
+                    open();
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SimpleConsoleApp.cs]: And every usage of a method and a field or property should be prefixed with `this.`.

File: CDP4-SDK-Samples/SimpleConsoleApp.cs [hunk 7]
```diff
@@ -0,0 +1,507 @@
...
+                }
+                default:
+                {
+                    Console.WriteLine("Unrecognized command.");
+                    break;
+                }
+            }
+        }
+
+        private async Task postPossibleFiniteStateListReorder()
```
Comment (alexatstariongroup) [inline CDP4-SDK-Samples/SimpleConsoleApp.cs]: Do please add simple xml documentation for methods and fields/properties. See othe c# repositories for the general look and feel.

**Review by alexatstariongroup — CHANGES_REQUESTED** [review]

**Review by alexatstariongroup — APPROVED** [review]
Good to merge!

**Pull request merged**